src/header/common/link.rs

```diff
@@ -0,0 +1,1027 @@
+    Current,
+    /// describedby.
+    DescribedBy,
+    /// edit.
+    Edit,
+    /// edit-media.
+    EditMedia,
+    /// enclosure.
+    Enclosure,
+    /// first.
+    First,
+    /// glossary.
+    Glossary,
+    /// help.
+    Help,
+    /// hub.
+    Hub,
+    /// index.
+    Index,
+    /// last.
+    Last,
+    /// latest-version.
+    LatestVersion,
+    /// license.
+    License,
+    /// next.
+    Next,
+    /// next-archive.
+    NextArchive,
+    /// payment.
+    Payment,
+    /// prev.
+    Prev,
+    /// predecessor-version.
+    PredecessorVersion,
+    /// previous.
+    Previous,
+    /// prev-archive.
+    PrevArchive,
+    /// related.
+    Related,
+    /// replies.
+    Replies,
+    /// section.
+    Section,
+    /// self.
+    RelationTypeSelf,
+    /// service.
+    Service,
+    /// start.
+    Start,
+    /// stylesheet.
+    Stylesheet,
+    /// subsection.
+    Subsection,
+    /// successor-version.
+    SuccessorVersion,
+    /// up.
+    Up,
+    /// versionHistory.
+    VersionHistory,
+    /// via.
+    Via,
+    /// working-copy.
+    WorkingCopy,
+    /// working-copy-of.
+    WorkingCopyOf,
+    /// ext-rel-type.
+    ExtRelType(String)
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Struct methods
+////////////////////////////////////////////////////////////////////////////////
+
+impl Link {
+    /// Create `Link` from a `Vec<LinkValue>`.
+    pub fn new(link_values: Vec<LinkValue>) -> Link {
+        Link { values: link_values }
+    }
+}
+
+#[allow(dead_code)]
+impl LinkValue {
+    /// Create `LinkValue` from URI-Reference.
+    pub fn new<T>(uri: T) -> LinkValue
+        where T: Into<Cow<'static, str>> {
+        LinkValue {
+            link: uri.into(),
+            rel: None,
+            anchor: None,
+            rev: None,
+            href_lang: None,
+            media_desc: None,
+            title: None,
+            title_star: None,
+            media_type: None,
+            link_extension: None,
+        }
+    }
+
+    /// Get the `LinkValue`'s value.
+    pub fn link(&self) -> &str {
+        self.link.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `rel` parameter(s).
+    pub fn rel(&self) -> Option<&Vec<RelationType>> {
+        self.rel.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `anchor` parameter.
+    pub fn anchor(&self) -> Option<&String> {
+        self.anchor.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `rev` parameter(s).
+    pub fn rev(&self) -> Option<&Vec<RelationType>> {
+        self.rev.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `hreflang` parameter(s).
+    pub fn href_lang(&self) -> Option<&Vec<LanguageTag>> {
+        self.href_lang.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `media` parameter(s).
+    pub fn media_desc(&self) -> Option<&Vec<MediaDesc>> {
+        self.media_desc.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `title` parameter.
+    pub fn title(&self) -> Option<&String> {
```

Comment:
Like the comment above, maybe we want to return `Option<&str>` instead. A plus to doing that is that it doesn't leak the internal type, so if we later changed `title` to hold a `Cow<'static, str>` or something, the API wouldn't need to change.